control-plane/config-entries/controllersv2/mesh_gateway_controller.go

```diff
@@ -83,19 +86,54 @@ func (r *MeshGatewayController) SetupWithManager(mgr ctrl.Manager) error {
 //  4. Role
 //  5. RoleBinding
 func (r *MeshGatewayController) onCreateUpdate(ctx context.Context, req ctrl.Request, resource *meshv2beta1.MeshGateway) error {
-	builder := gateways.NewMeshGatewayBuilder(resource)
+	//fetch gatewayclassconfig
+	gcc, err := r.getGatewayClassConfigForGateway(ctx, resource)
+	if err != nil {
+		r.Log.Error(err, "unable to get gatewayclassconfig for gateway: %s gatewayclass: %s", resource.Name, resource.Spec.GatewayClassName)
+		return err
+	}
+
+	builder := gateways.NewMeshGatewayBuilder(resource, r.GatewayConfig, gcc)
 
 	upsertOp := func(ctx context.Context, _, object client.Object) error {
 		_, err := controllerutil.CreateOrUpdate(ctx, r.Client, object, func() error { return nil })
 		return err
 	}
 
-	err := r.opIfNewOrOwned(ctx, resource, &corev1.ServiceAccount{}, builder.ServiceAccount(), upsertOp)
+	err = r.opIfNewOrOwned(ctx, resource, &corev1.ServiceAccount{}, builder.ServiceAccount(), upsertOp)
 	if err != nil {
+		return fmt.Errorf("unable to create service account: %w", err)
+	}
+
+	// TODO NET-6393 NET-6395
+
+	//Create deployment
+
+	mergeDeploymentOp := func(ctx context.Context, existingObject, object client.Object) error {
+		existingDeployment, ok := existingObject.(*appsv1.Deployment)
+		if !ok && existingDeployment != nil {
```

Comment:
What's the purpose of the nil check here? Is it possible to get a non-nil existing deployment here?

Reply:
Yes. On creation built will be non nil and existing will be nil.

Reply:
Additionally when existing is nil the ok will be false.

Reply:
Should this be nil-checking `existingObject` though instead of `existingDeployment`?

Reply:
TIL Checked in the Go playground, turns out you do get a value for `existingDeployment` here if non-nil